tests/test_middleware.py

```diff
@@ -0,0 +1,119 @@
+from werkzeug.test import Client
+
+import pytest
+
+from freezeyt.middleware import Middleware
+from freezeyt.util import WrongMimetypeError
+
+from testutil import APP_NAMES, context_for_test
+
+def urls_from_expected_dict(expected_dict, prefix=''):
+    """Generate URLs from an `expected_dict` found in tests
+
+    See test_urls_from_expected_dict for examples
+    """
+    for name, value in expected_dict.items():
+        if isinstance(value, bytes):
+            if name == 'index.html':
+                yield prefix + '/'
+            else:
+                yield prefix + '/' + name
+        else:
+            new_prefix = prefix + '/' + name
+            yield from urls_from_expected_dict(value, prefix=new_prefix)
+
+def test_urls_from_expected_dict():
+    """Test the test helper, urls_from_expected_dict"""
+    assert sorted(urls_from_expected_dict(
+        {
+            'index.html': b'...',
+            'other.html': b'...',
+            'extra': {
+                'index.html': b"...",
+            },
+            'extra2': {'index.html': b'Extra page 2'},
+            'extra3': {'index.html': b'Extra page 3'},
+            'extra4': {'index.html': b'Extra page 4'},
+            'extra5': {'index.html': b'Extra page 5'},
```

Comment:
Why do we test these three extra (3, 4, 5)? Have these tests some extra value for us or we can omitted them?